source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -925,10 +925,14 @@ namespace
                     {
                         massign->setExclusive(rtMatAssign.getExclusive().getValue().asBool());
                         auto iter = rtMatAssign.getCollection().getTargets();
-                        massign->setCollectionString((*iter).getName());
+                        if (!iter.isDone()) {
+                            massign->setCollectionString((*iter).getName());
+                        }
```

Comment:
Can happen, especially in interactive Arnold mode in Maya, that a materialassign is saved while not yet having a material or a collection.